src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -86,6 +86,84 @@ internal static class ParenthesizedExpressionSyntaxExtensions
+                        {
+                            break;
+                        }
+
+                        previousArgument = tupleArgument;
+                    }
+
+                    if (previousArgument is not null)
+                    {
+                        var previousArgumentExpression = previousArgument.Expression;
+
+                        if (previousArgumentExpression is ParenthesizedExpressionSyntax parenthesized)
+                        {
+                            previousArgumentExpression = parenthesized.Expression;
+                        }
+
+                        // Now checking previous tuple argument not to be "... < SomeIdentifier"
+                        if (previousArgumentExpression is not null &&
+                            previousArgumentExpression is BinaryExpressionSyntax previousArgumentAsBynaryExpression &&
+                            previousArgumentAsBynaryExpression.IsKind(SyntaxKind.LessThanExpression) &&
+                            previousArgumentAsBynaryExpression.Right.IsKind(SyntaxKind.IdentifierName))
+                        {
+                            return false;
+                        }
+                    }
+                }
+
+                // Opposite of the first check: "... < SomeIdentifier"
+                if (binaryExpression.IsKind(SyntaxKind.LessThanExpression) && binaryExpression.Right.IsKind(SyntaxKind.IdentifierName))
+                {
+                    ArgumentSyntax? nextArgument = null;
+                    var nextArgumentFound = false;
+
+                    foreach (var tupleArgument in tupleExpression.Arguments)
+                    {
+                        if (nextArgumentFound)
+                        {
+                            nextArgument = tupleArgument;
+                            break;
+                        }
+
+                        nextArgumentFound = tupleArgument == currentArgument;
+                    }
+
+                    if (nextArgument is not null)
+                    {
+                        var nextArgumentExpression = nextArgument.Expression;
+
+                        if (nextArgumentExpression is ParenthesizedExpressionSyntax parenthesized)
+                        {
+                            nextArgumentExpression = parenthesized.Expression;
+                        }
+
+                        // Now checking next tuple argument not to be "SomeIdentifier > ..."
+                        if (nextArgumentExpression is not null &&
+                            nextArgumentExpression is BinaryExpressionSyntax nextArgumentAsBynaryExpression &&
+                            nextArgumentAsBynaryExpression.IsKind(SyntaxKind.GreaterThanExpression) &&
+                            nextArgumentAsBynaryExpression.Left.IsKind(SyntaxKind.IdentifierName))
+                        {
+                            return false;
+                        }
+                    }
+                }
+            }
```

Comment:
So the issue reported is not a realistic case.  It was a user intentionally trying to break things.  As such, we'd only be open for a fix here if it's extremely minimal in size/scope.  The above code is *waaaaay* too complex to have in teh product for a non-realistic case that people will not actually hit in practice.

Reply:
It's a bunch of code, that is just checking for this specific case. If you open a corresponding issue it becomes relatively easy to read it. I don't think it is possible to fix the issue in less code size, so if you are sure, that you are gonna reject this contribution, please close the issue as well.

And if the issue is so artificial, why haven't you closed it, when you first saw it then btw?

Reply:
> It's a bunch of code, that is just checking for this specific case. 

Our principle here is that we want the amoutn of code to be commensurate to the amount of value it provides.  In this case this is not a real user scenario.  This is only being hit by someone in particular trying to see how features work in response to this language ambiguity.  As such, it's a lot of code being added to not actually improve things for a realistic section of our user base for realistic coding patterns.  As such, the amount of code would need to be correspondingly tiny to be ok here.

Reply:
>  I don't think it is possible to fix the issue in less code size, so if you are sure, that you are gonna reject this contribution, please close the issue as well.

I would not reject a contribution that was tiny and addressed the issue.  I'm not sure if such a contribution is possible as i have not thought about it at all.

Reply:
> And if the issue is so artificial, why haven't you closed it, when you first saw it then btw?

Because the issue does exist.  And if a suitable solution can be found for it, then that would be ok.  It's not a case of "we will not ever take a solution here".  It's "we'll take an appropriately sized solution here if possible".

Reply:
If you're available to chat, woudl love to discuss this further.  Thanks! :)